src/test/ui/typeck/issue-86721-return-expr-ice.rs

```diff
@@ -0,0 +1,17 @@
+// Regression test for the ICE described in #86721.
+
+// revisions: rev1 rev2
```

Comment:
If we want to check that no ICE occurs having those two cases in a single test with no revisions should be fine

Reply:
I thought so, too, but for some reason, the type checker aborts after the first error here, so I have had to split this into two revisions to make sure both traits are type checked.